mbed_devices/_internal/windows/windows_identifier.py

```diff
@@ -0,0 +1,68 @@
+"""Defines a Windows identifier."""
+from typing import NamedTuple, Any, Optional
+
+from mbed_devices._internal.windows.component_descriptor_utils import is_undefined_value
+
+INSTANCE_ID_CHARACTER = "&"
+
+
+class WindowsUID(NamedTuple):
+    """Definition of a Windows Universal ID."""
+
+    UID: str  # Processed string which may be the closest to the Universal ID.
+    raw_uid: str  # The String from which the UID was derived. Raw form as it can be a
+    # combination of other elements e.g. 8&3432E179&0&000
+    serial_number: Optional[str]  # a serial number
+
+    def __eq__(self, other: Any) -> bool:
+        """Defines the equality checker."""
+        if not other or not isinstance(other, WindowsUID):
+            return False
+        if (
+            self.UID == other.UID
+            or self.raw_uid == other.raw_uid
+            or self.serial_number == other.serial_number
+            or self.serial_number == other.UID
+            or self.UID == other.serial_number
+        ):
+            return True
+        # Due to the complexity of determining the UID on Windows,
+        # we can assume that we are dealing with the same ID if IDs are subsets of each other.
+        return str(self.UID).startswith(str(other.UID)) or str(other.UID).startswith(str(self.UID))
+
+    @property
+    def actual_serial_number(self) -> str:
+        """Determines the actual serial number of a device from the different components."""
```

Comment:
Could you add more documentation/examples here? I think readers of this might not know what "instance ID" is. Also worth explaining what "different components" are.

Reply:
Hopefully I have been more clear